src/main/java/org/jitsi/jigasi/transcription/GoogleCloudStreamingRecognitionSession.java

```diff
@@ -0,0 +1,692 @@
+            // nothing new was said, but there should be an error in the message
+            // so this is never supposed to happen
+            if(alternatives.isEmpty())
+            {
+                logger.warn(debugName + ": received a list of alternatives which" +
+                                    " was empty");
+                requestManager.terminateCurrentSession();
+                return;
+            }
+
+            handleResult(finalResult);
+
+            requestManager.terminateCurrentSession();
+        }
+
+        /**
+         * Get whether a {@link StreamingRecognizeResponse} has an
+         * {@link StreamingRecognizeResponse.SpeechEventType} of
+         * {@link StreamingRecognizeResponse.SpeechEventType#
+         * END_OF_SINGLE_UTTERANCE}
+         *
+         * @param message the message to check
+         * @return true if the message has the eventType
+         * {@link StreamingRecognizeResponse.SpeechEventType
+         * #END_OF_SINGLE_UTTERANCE}, false otherwise
+         */
+        private boolean isEndOfSingleUtteranceMessage(
+                StreamingRecognizeResponse message)
+        {
+            return message.getSpeechEventType().
+                    equals(StreamingRecognizeResponse.SpeechEventType.
+                                   END_OF_SINGLE_UTTERANCE);
+        }
+
+        /**
+         * Handle a single {@link StreamingRecognitionResult} by creating
+         * a {@link TranscriptionResult} based on the result and notifying all
+         * all registered {@link TranscriptionListener}s
+         *
+         * @param result the result to handle
+         */
+        private void handleResult(StreamingRecognitionResult result)
+        {
+            List<SpeechRecognitionAlternative> alternatives
+                    = result.getAlternativesList();
+
+            if(alternatives.isEmpty())
+            {
+                return;
+            }
+
+            TranscriptionResult transcriptionResult = new TranscriptionResult(
+                    null,
+                    this.messageID,
+                    !result.getIsFinal(),
+                    this.languageTag,
+                    result.getStability());
+
+            for(SpeechRecognitionAlternative alternative : alternatives)
+            {
+                transcriptionResult.addAlternative(
+                        new TranscriptionAlternative(
+                                alternative.getTranscript(),
+                                alternative.getConfidence()));
+            }
+
+            sent(transcriptionResult);
+        }
+
+        @Override
+        public void onError(Throwable t)
+        {
+            logger.warn(debugName + ": received an error from the Google Cloud API", t);
+            requestManager.terminateCurrentSession();
+        }
+
+        @Override
+        public void onCompleted()
+        {
+            for(TranscriptionListener listener : requestManager.getListeners())
```

Comment:
I think it would be valuable to trace what listeners have been to stop or whether whether the list of listeners is empty